tools/tidb-binlog-cluster.md

```diff
@@ -0,0 +1,566 @@
+### 下载 tidb-ansible
+
+以 tidb 用户登录中控机并进入 `/home/tidb` 目录，使用以下命令 tidb-ansible `new-tidb-binlog` 分支，默认的文件夹名称为 tidb-ansible。
+
+```
+$ git clone -b new-tidb-binlog https://github.com/pingcap/tidb-ansible.git
+```
+
+### 部署 Pump
+#### 修改 tidb-ansible/inventory.ini 文件
+
+1. 设置 `enable_binlog = True`，表示 TiDB 集群开启 binlog。
+
+```
+## binlog trigger
+enable_binlog = True
+```
+
+2. 为 `pump_servers` 主机组添加部署机器 IP。
+
+```
+## Binlog Part
+[pump_servers]
+172.16.10.72
+172.16.10.73
+172.16.10.74
+```
+
+默认 pump 保留 5 天数据，如需修改可修改 `tidb-ansible/conf/pump.yml` 文件中 `gc` 变量值，并取消注释，如修改为 7。
+
+```
+global:
+  # a integer value to control expiry date of the binlog data, indicates for how long (in days) the binlog data would be stored. 
+  # must bigger than 0
+  gc: 7
+```
+
+请确保部署目录有足够空间存储 binlog，详见：[部署目录调整](../op-guide/ansible-deployment.md#部署目录调整)，也可为 pump 设置单独的部署目录。
+
+```
+## Binlog Part
+[pump_servers]
+pump1 ansible_host=172.16.10.72 deploy_dir=/data1/pump
+pump2 ansible_host=172.16.10.73 deploy_dir=/data1/pump
+pump3 ansible_host=172.16.10.74 deploy_dir=/data1/pump
+```
+
+#### 部署并启动 TiDB 集群。
+
+使用 ansible 部署 TiDB 集群的具体方法参考 [TiDB Ansible 部署方案](../op-guide/ansible-deployment.md)，开启 binlog 后默认会部署和启动 pump 服务。
+
+#### 查看 Pump 服务状态
+
+使用 binlogctl 查看 pump 服务状态，pd-urls 参数请替换为集群 PD 地址，结果 State 为 online 表示 pump 启动成功。
+
+```
+$ cd /home/tidb/tidb-ansible
+$ resources/bin/binlogctl -pd-urls=http://172.16.10.72:2379 -cmd pumps
+2018/09/21 16:45:54 nodes.go:46: [info] pump: &{NodeID:ip-172-16-10-72:8250 Addr:172.16.10.72:8250 State:online IsAlive:false Score:0 Label:<nil> MaxCommitTS:0 UpdateTS:403051525690884099}
+2018/09/21 16:45:54 nodes.go:46: [info] pump: &{NodeID:ip-172-16-10-73:8250 Addr:172.16.10.73:8250 State:online IsAlive:false Score:0 Label:<nil> MaxCommitTS:0 UpdateTS:403051525703991299}
+2018/09/21 16:45:54 nodes.go:46: [info] pump: &{NodeID:ip-172-16-10-74:8250 Addr:172.16.10.74:8250 State:online IsAlive:false Score:0 Label:<nil> MaxCommitTS:0 UpdateTS:403051525717360643}
+```
+
+### 部署 Drainer
+#### 获取 initial_commit_ts 
+
+使用 binlogctl 工具生成 Drainer 初次启动所需的 tso 信息，命令：
+
+```
+$ cd /home/tidb/tidb-ansible
+$ resources/bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd generate_meta
+INFO[0000] [pd] create pd client with endpoints [http://192.168.199.118:32379]
+INFO[0000] [pd] leader switches to: http://192.168.199.118:32379, previous:
+INFO[0000] [pd] init cluster id 6569368151110378289
+2018/06/21 11:24:47 meta.go:117: [info] meta: &{CommitTS:400962745252184065}
+```
+ 
+该命令会生成一个文件 `{data-dir}/savepoint`，该文件中包含 tso，用该 tso 作为 drainer 初次启动使用的 initial-commit-ts 参数的值.
+
+#### 全量数据的备份与恢复
+如果下游为 MySQL/TiDB, 需要保证数据的完整性，在 Drainer 启动前（Pump 运行后十分钟左右）进行数据的全量备份和恢复。
+
+推荐使用 mydumper 备份 TiDB 的全量数据，再使用 loader 将备份数据导入到下游。具体使用方法参考：[备份与恢复](https://github.com/pingcap/docs-cn/blob/master/op-guide/backup-restore.md)
+
+
+#### 修改 tidb-ansible/inventory.ini 文件
+
+为 `drainer_servers` 主机组添加部署机器 IP，initial_commit_ts 请设置为获取的 initial_commit_ts，仅用于 drainer 第一次启动。
```

Comment:
drainer -> Drainer 

**Please update all "drainer" to "Drainer" where necessary in this PR.**